### ISSUE TITLE
Integrate model for global identity assignment to unassigned tracklets

### ISSUE THREAD

**Issue by emdavis02** [talmolab/sleap-mot#28]
## Feature Description

Currently, the tracking pipeline creates tracklets but does not assign them to a global identity if the individual is not identifiable at any point during the tracklet. We need to integrate a model that can take these unassigned tracklets and match them to a set of known global identities based on the tracklets that have been assigned to a global identity. This will take place as the **last** step of the tracking algorithm.

### Proposed Changes
**Step 1: Write function to extract features from a pose to feed the model**
It will return a feature array including position and appearance data
```python
def extract_pose_features(instance: sio.PredictedInstance) -> np.ndarray:
    """Extract pose features from an instance for HMM input.
    
    Args:
        instance: The instance to extract features from.
        
    Returns:
        np.ndarray: Flattened feature vector containing:
            - Centroid coordinates (x, y)
            - Patches pixel values
    """
    centroid = get_centroid(instance)
    patches = get_patches(instance)
    
    # Concatenate all features
    features = np.concatenate([
        centroid,  
        patches
    ])
    
    return features
```

**Step 2: Write function to train a model from these features**
Model type is a parameter
```python
def train_model_from_tracked_instances(labels, model_type):
    """
    Args:
        labels: contains instances with global track IDs
        model_type: string (e.g., "HMM", "SVM", etc.)
    """
    # Group features by track ID
    sequences = []
    lengths = []
    for inst in tracked_instances(labels):
        features = extract_pose_features(inst)
        add_to_sequence(inst.track.name, features)

    # Flatten and scale features
    X = stack_all_sequences(sequences)
    X_scaled = scale_features(X)

    # Initialize and train model
    model = init_model(model_type)
    model.fit(X_scaled, lengths)

    return model
```

**Step 3: Write a function that takes in the trained model and the labels and IDs unidentified tracklets**
Include logic such as if a global ID appears during any frame of an tracklet, this ID is not available to assign to said tracklet
```python
def infer_and_assign_untracked_identities(trained_model, labels):
    tracklets = group_tracklets(labels)

    for track_id, instances in tracklets.items():
        if track_id not in global_track_ids:
            # Sort by frame and extract features
            frames, sorted_instances = sort_and_extract(instances)
            features = [extract_pose_features(inst) for inst in sorted_instances]
            features_scaled = scale_and_clean(features)

            if features_scaled is None:
                continue  # Skip if invalid

            # Predict state sequence from model
            state_seq = trained_model.predict(features_scaled)
            likely_id = majority_vote(state_seq)

            # Map state to global identity
            global_id = map_state_to_global_id(likely_id)

            # Check if ID is already used in any frame of this tracklet
            if is_id_used_in_frames(global_id, frames, labels):
                # Try alternative IDs
                global_id = find_available_id(global_track_ids, frames, labels)
                if global_id is None:
                    warn_no_available_id(track_id)
                    continue

            # Assign global identity to all instances in tracklet
            assign_id_to_instances(global_id, sorted_instances)
            log_assignment(track_id, global_id)
```

### Questions to Consider

1. How to handle tracklets that don't match any global ID?
  a. generate a new global ID if none are available
3. What confidence threshold to use for assignments?
